04.ls/ls.rb

```diff
@@ -33,4 +33,118 @@ def split_files(files)
   divided_files
 end
 
+FILE_TYPE = {
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '-',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+
+PERMISSION = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx'
+}.freeze
+
+def load_l_option(path, file_names)
+  stat_files = file_names.map { |file_name| File.lstat("#{path}/#{file_name}") }
+  path_to_stat = stat_files.map.with_index { |stat, idx| ["#{path}/#{file_names[idx]}", stat] }.to_h
```

Comment:
`path_to_stat.values`が`stat_files`と同じになるはずなので、

```suggestion
  path_to_stat = file_names.to_h { ... }
```

みたいな形で、いきなりハッシュを作ってしまっていいのでは？と思いました。

Reply:
1行に収める形に変更しました。

```ruby
# 変更前
stat_files = file_names.map { |file_name| File.lstat("#{path}/#{file_name}") }
path_to_stat = stat_files.map.with_index { |stat, idx| ["#{path}/#{file_names[idx]}", stat] }.to_h

# 変更後
path_to_stat = file_names.to_h { |file_name| [File.join(path, file_name), File.lstat(File.join(path, file_name))] }
```